src/common/utility.ts

```diff
@@ -14,6 +15,33 @@ import {
 } from './constants'
 import { localize } from 'components/localization'
 
+export const epochToMoment = (epoch: number) => moment.unix(epoch).utc()
+export const toMoment = (value?: moment.MomentInput): moment.Moment => {
+    if (!value) return moment().utc() // returns 'now' moment object
+    if (
+        value instanceof moment &&
+        (value as moment.Moment).isValid() &&
+        (value as moment.Moment).isUTC()
+    )
+        return value as moment.Moment // returns if already a moment object
+    if (typeof value === 'number') return epochToMoment(value) // returns epochToMoment() if not a date
+
+    if (/invalid/i.test(moment(value).toString())) {
+        const today_moment = moment()
+        const days_in_month = today_moment.utc().daysInMonth()
+        const value_as_number = moment.utc(value, 'DD MMM YYYY').valueOf() / (1000 * 60 * 60 * 24)
+        return value_as_number > days_in_month
+            ? moment.utc(today_moment.add(value as string | number, 'd'), 'DD MMM YYYY')
+            : moment.utc(value, 'DD MMM YYYY') // returns target date
+    }
+    return moment.utc(value)
```

Comment:
```suggestion
   if (!value) return moment().utc(); // returns 'now' moment object
    if (moment.isMoment(value) && value.isValid() && value.isUTC())
        return value; // returns if already a moment object
    if (typeof value === 'number') return epochToMoment(value); // returns epochToMoment() if not a date

    if (/invalid/i.test(moment(value).toString())) {
        const today_moment = moment();
        const days_in_month = today_moment.utc().daysInMonth();
        const value_as_number = moment.utc(value, 'DD MMM YYYY').valueOf() / (1000 * 60 * 60 * 24);
        if (typeof value === 'string' || typeof value === 'number') {
            return value_as_number > days_in_month
                ? moment.utc(today_moment.add(value, 'd'), 'DD MMM YYYY')
                : moment.utc(value, 'DD MMM YYYY'); // returns target date
        }
    }
    return moment.utc(value);
```

Reply:
@yaswanth-deriv, if you use `moment.isMoment(value)`, then you don't need to `(value as moment.Moment).isValid()`. Also, is it possible to extract this into a separate function so that is can be reused?

Reply:
I will change it.Thanks